### PR TITLE
Added Webhook / updated method 'GetWebhooks' to return webhookList object

### PR DESCRIPTION
**Pull request by haseebmukhtar**

### CODE REVIEW

**Comment by XiaoFaye** [comment]
Hello mate, I only see a .gitignore file, have you missed something? :)

**Comment by haseebmukhtar** [comment]
Hi
I have just commit again with other files, can you please check now.

Thanks

File: WooCommerce/Webhook.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        [DataMember(EmitDefaultValue = false)]
+        public string secret { get; set; }
+
+
+        /// <summary>
+        /// UTC DateTime when the webhook was created
+        /// read-only
+        /// </summary>
+        [DataMember(EmitDefaultValue = false)]
+        public string created_at { get; set; }
```
Comment (XiaoFaye) [inline WooCommerce/Webhook.cs]: We can use DateTime as the data type for created_at.

File: WooCommerce/Webhook.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        /// <summary>
+        /// UTC DateTime when the webhook was last updated
+        /// read-only
+        /// </summary>
+        [DataMember(EmitDefaultValue = false)]
+
+        public string updated_at { get; set; }
```
Comment (XiaoFaye) [inline WooCommerce/Webhook.cs]: We can use DateTime as the data type for updated_at.

File: WooCommerceNET.sln [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿
```
Comment (XiaoFaye) [inline WooCommerceNET.sln]: This file might not be working on others' machine.

File: WooCommerce/WCObject.cs [hunk 4]
```diff
@@ -347,11 +347,11 @@ public WCObject(RestAPI api)
 
         #region "webhooks..."
 
-        public async Task<string> GetWebhooks(Dictionary<string,string> parms = null)
+        public async Task<WebhookList> GetWebhooks(Dictionary<string,string> parms = null)
         {
             string json = await API.SendHttpClientRequest("webhooks", RequestMethod.GET, string.Empty, parms);
             json = json.Substring(json.IndexOf('['), json.Length - json.IndexOf('[') - 1);
```
Comment (XiaoFaye) [inline WooCommerce/WCObject.cs]: The below code will work for both cases.
`json = json.Substring(json.IndexOf(':') + 1, json.Length - json.IndexOf(':') - 2);`

File: WooCommerce/WCObject.cs [hunk 5]
```diff
@@ -360,11 +360,11 @@ public WCObject(RestAPI api)
             return int.Parse(json.Substring(json.IndexOf(':') + 1, json.IndexOf('}') - json.IndexOf(':') - 1).Trim('"'));
         }
 
-        public async Task<string> GetWebhook(int id, Dictionary<string,string> parms = null)
+        public async Task<Webhook> GetWebhook(int id, Dictionary<string,string> parms = null)
         {
             string json = await API.SendHttpClientRequest("webhooks/" + id.ToString(), RequestMethod.GET, string.Empty, parms);
             json = json.Substring(json.IndexOf(":{") + 1, json.Length - json.IndexOf(":{") - 2);
```
Comment (XiaoFaye) [inline WooCommerce/WCObject.cs]: The below code will work for both cases.
`json = json.Substring(json.IndexOf(':') + 1, json.Length - json.IndexOf(':') - 2);`

**Comment by haseebmukhtar** [comment]
Hi,
Thanks for guidance and help, I am new on github.

For webhooks the created date and updated date is returning in very strange format ("-0001-11-30T00:00:00Z") and Json converter cannot parse if we set the datatype datetime.

JSON results:

{"webhooks":[{"id":13,"name":"Webhook created on May 13, 2016 @ 05:33 PM","status":"active","topic":"product.created","resource":"product","event":"created","hooks":["woocommerce_process_product_meta","woocommerce_api_create_product"],"delivery_url":"","created_at":"-0001-11-30T00:00:00Z","updated_at":"-0001-11-30T00:00:00Z"}]}


Any ideas or suggestion please.

**Comment by XiaoFaye** [comment]
It seems the created_at_gmt and updated_at_gmt fields in the wp_posts table are always set to 0000-00-00T00:00:00Z for webhooks, which will be returned by the REST api. I have confirmed this behavior with the latest version of woocommerce.  I will have a further look and see if this is a bug.

**Comment by haseebmukhtar** [comment]
Just tested again, seems a bug in WooCommerce, should I leave it as string for now.

**Comment by haseebmukhtar** [comment]
requesting new pull

**Pull request closed without merging**